batch_job/src/work_queue_factory.c

```diff
@@ -718,7 +722,7 @@ int read_config_file(const char *config_file) {
 
 	assign_new_value(new_factory_timeout_option, factory_timeout, factory-timeout, int, JX_INTEGER, integer_value)
 
-	assign_new_value(new_tasks_per_worker, tasks_per_worker, tasks-per-worker, double, JX_INTEGER, integer_value)
+	assign_new_value(new_tasks_per_worker, tasks_per_worker, tasks-per-worker, int, JX_INTEGER | JX_DOUBLE, integer_value)
```

Comment:
I think we can safely convert all numeric JX_DOUBLE to ints after a ceil() around line 676. We will need another if to check that it is a double and do the corresponding ceil.